src/index.js

```diff
@@ -57,8 +60,22 @@ export default function rehypePrettyCode(options = {}) {
     getHighlighter = shikiHighlighter,
   } = options;
 
-  // Cache highlighters per unified processor
-  const highlighterCache = new Map();
+  const optionsHash = hashObj(
+    {
+      theme,
+      tokensMap,
+      onVisitLine,
+      onVisitHighlightedLine,
+      onVisitHighlightedWord,
+      getHighlighter,
+    },
+    {algorithm: 'sha1'}
+  );
```

Comment:
What I'm wondering is how this actually acts when passed function values? (e.g. referential stability vs. stringified representation, etc). Maybe add tests for how this functions?

Reply:
The hash remains the same, the group of tests `with fixtures` is where I'm indirectly testing for all of that by extracting `getHighlighter` outside. `runFixture` passes new functions each time and creates a new highlighter on each run, just like the problem we're trying to solve.